lib/verification.js

```diff
@@ -23,17 +23,35 @@ export async function verifyRequest(rawBody, signature, key) {
 
 /** @type {import('..').FetchVerificationKeysInterface} */
 export async function fetchVerificationKeys(
-  { token = "", request = defaultRequest } = { request: defaultRequest }
+  { token = "", request = defaultRequest, cache = { id: "", keys: [] } } = { request: defaultRequest }
 ) {
-  const { data } = await request("GET /meta/public_keys/copilot_api", {
-    headers: token
+    try {
+    const headers = token
       ? {
           Authorization: `token ${token}`,
         }
-      : {},
-  });
+      : {};
+
+    if (cache.id) headers["If-None-Match"] = cache.id;
+
+    const response = await request("GET /meta/public_keys/copilot_api", {
+      headers,
+    });
+
+    if (response.status === 304) {
+      return cache;
+    }
+
+    const cacheId = response.headers.etag || "";
+    return { id: cacheId, keys: response.data.public_keys };
+  } catch (error) {
+    if (error.status === 304) {
```

Comment:
When I originally [wrote up the tests](https://github.com/copilot-extensions/preview-sdk.js/pull/81/files#diff-888235d23d4180ab42cebcaa7b47a7d3ec586a7290a7e5acaf4d05fe5ca9eb04R188-R237), I thought this would "just work": however, I saw that it failed because an error was thrown.

Re-reviewing my work, the only place where an error would be thrown would be on the request for the public keys](https://github.com/copilot-extensions/preview-sdk.js/pull/81/files#diff-7b02190766e81ebccdc674c1257d3ceead3e9c97c6d0968a5856749625cb13f3R37).

I researched `octokit/request.js` and looked into [what happens when a `status: 304` is returned](https://github.com/octokit/request.js/blob/bec87c90d08449815ed8cac4f86eb3d082ed5f9c/src/fetch-wrapper.ts#L124-L131): it turns out that it captures the data, but seems to go on to throw a new `RequestError` with the full request-response pair. 😅 

I'm sure there's a reason (that I'm unaware of) for why it's written in this way, so I chose to catch the error and check its status; should it be a `304`, we go on to return the `cache`. Otherwise (for say a `4xx` or `5xx` error), we'll throw the error. 

Happy to make updates to the error handling here with some more guidance! 😉 👼

Reply:
`@octokit/request` is throwing an error in case of a `304` because it cannot give you the response body. Throwing an error was the only way that I could think of to give developers a way to handle the case when a conditional request matches, meaning the response is 304 without any body. In that case it's up to the developer to provide the body instead.

I feel a proper high-level API for conditional request is really missing from the Octokit SDK, I never got around to give it more thought. 

All this to say, the way you implemented it is the correct way.